### PR TITLE
Theme designer v9

### PR DESCRIPTION
**Pull request by timbellay**
## Previous Behavior
- N/A, adding some cleanup items for Fluent V9

## New Behavior
- Collapsed app state into single state with reducer. Fixed infinite state updating bug.
- Simplified flow of app using the left nav and steps for user to generate palette.
- Updated sticker sheet to show missing v9 components
- Created a test object and a test system for future tests and clarity of test results. 
- Reduced contrast tests to handle the main two tests only: colorBrandForeground1: ['colorNeutralBackground1', 4.5] and colorBrandForeground2: ['colorBrandBackground2', 4.5] for simplicity.
- Added contrast ratios to dropdown in the token selection menu of failed tests so users can know the effect of changing a token value on the test.
- Change curve snapping points from [0, 50, 100] to [start, middle, end] based on Arman's spec and hard coded this in hueToSnappingPointsMap.
- Use brand[110] instead of brand[100] and brand[120] instead of brand[110] for dark theming per guidelines
- Minor GUI improvements

## Related Issue(s)
- hue-shift in the palette generation: https://github.com/microsoft/fluentui/issues/27860

- Fixes #
N/A

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/06cd515eda59a3078bfe32f9cc7c1f281cd20208'>06cd515eda59a3078bfe32f9cc7c1f281cd20208</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-components/theme-designer/src/Context/ThemeDesignerContext.tsx [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+      return {
+        ...state,
+      };
+  }
+};
+
+export const useThemeDesigner = () => {
+  const context = React.useContext(ThemeDesignerContext);
+  if (context === undefined) {
+    throw new Error('useWizard must be used within a WizardProvider');
```
Comment (TristanWatanabe) [inline packages/react-components/theme-designer/src/Context/ThemeDesignerContext.tsx:185]: What is `useWizard`? I couldn't find any other reference of that in this PR

File: packages/react-components/theme-designer/src/colors/palettes.ts [hunk 2]
```diff
@@ -14,12 +14,69 @@ import { CurvedHelixPath, Palette, Vec3 } from './types';
...
-  for (let i = 0; i < n; i++) {
-    result[i] = min + delta * i;
+// function getLinearSpace(min: number, max: number, n: number) {
+//   const result = [];
+//   const delta = (max - min) / n;
+//   for (let i = 0; i < n; i++) {
+//     result[i] = min + delta * i;
+//   }
+//   return result;
+// }
```
Comment (TristanWatanabe) [inline packages/react-components/theme-designer/src/colors/palettes.ts:24]: should this be deleted?

File: packages/react-components/theme-designer/src/colors/palettes.ts [hunk 3]
```diff
@@ -14,12 +14,69 @@ import { CurvedHelixPath, Palette, Vec3 } from './types';
...
+    hueToSnappingPointsMap[hue][0] * 100,
+    hueToSnappingPointsMap[hue][1] * 100,
+    hueToSnappingPointsMap[hue][2] * 100,
+  ];
+  //   console.log(`hue ${hue} range, ${range}`);
+  return range;
+};
+
+const pointsForKeyColor = (keyColor: string, range: number[], centerPoint: number): number[] => {
+  //   const rgb = hex_to_sRGB(keyColor);
```
Comment (TristanWatanabe) [inline packages/react-components/theme-designer/src/colors/palettes.ts:38]:
```suggestion
  //   const rgb = hex_to_sRGB(keyColor);
```

File: packages/react-components/theme-designer/src/colors/palettes.ts [hunk 4]
```diff
@@ -14,12 +14,69 @@ import { CurvedHelixPath, Palette, Vec3 } from './types';
...
+  const hue = hexToHue(keyColor);
+  const center = hueToSnappingPointsMap[hue][1] * 100;
+  const linear = linearInterpolationThroughPoint(range[0], range[1], center, 16);
+  //   const linear = linearInterpolationThroughPoint(range[0], range[1], (range[1]-range[0])/2, 16);
```
Comment (TristanWatanabe) [inline packages/react-components/theme-designer/src/colors/palettes.ts:42]:
```suggestion

```

File: packages/react-components/theme-designer/src/colors/palettes.ts [hunk 5]
```diff
@@ -39,21 +96,22 @@ const getLogSpace = (min: number, max: number, n: number) => {
...
+  const snappingPoints = snappingPointsForKeyColor(keyColor);
   const paletteShades = [];
-
-  const logLightness = getLogSpace(Math.log10(range[0]), Math.log10(range[1]), nShades);
-
-  const linearLightness = getLinearSpace(range[0], range[1], nShades);
-
+  const range = [snappingPoints[0], snappingPoints[2]];
+  const logLightness = getLogSpace(Math.log10(0), Math.log10(100), nShades);
+  //   const oldLinearLightness = getLinearSpace(range[0], range[1], nShades); // older attempt
```
Comment (TristanWatanabe) [inline packages/react-components/theme-designer/src/colors/palettes.ts:110]:
```suggestion

```

File: packages/react-components/theme-designer/src/colors/palettes.ts [hunk 6]
```diff
@@ -152,25 +206,25 @@ export function curvePathFromPalette({ keyColor, darkCp, lightCp, hueTorsion }:
...
+// export function cssGradientFromCurve(
+//   curve: CurvedHelixPath,
+//   nShades = 16,
+//   range = [0, 100],
+//   linearity = defaultLinearity,
+//   curveDepth = 24,
+// ) {
+//   const hexes = paletteShadesToHex(paletteShadesFromCurve(curve, nShades, range, linearity, curveDepth));
+//   return `linear-gradient(to right, ${hexes.join(', ')})`;
+// }
```
Comment (TristanWatanabe) [inline packages/react-components/theme-designer/src/colors/palettes.ts:218]: Should these be deleted as well?

File: packages/react-components/theme-designer/src/components/ColorTokens/AccessiblePairs.ts [hunk 7]
```diff
@@ -7,98 +7,130 @@ export type AccessiblePairs = Record<string, [string, number][]>;
...
-  colorCompoundBrandForeground1Pressed: [
-    ['colorNeutralBackground1', 3],
-    ['colorNeutralBackground1Pressed', 3],
-    ['colorTransparentBackgroundPressed', 3],
-    ['colorSubtleBackgroundPressed', 3],
-  ],
+  //   colorNeutralForeground2BrandHover: [
+  //     ['colorTransparentBackgroundHover', 4.5],
+  //     ['colorSubtleBackgroundHover', 3],
+  //   ],
```
Comment (TristanWatanabe) [inline packages/react-components/theme-designer/src/components/ColorTokens/AccessiblePairs.ts:13]: Were these meant to be commented out?

File: packages/react-components/theme-designer/src/components/Nav/Nav.tsx [hunk 8]
```diff
@@ -38,22 +36,25 @@ export const Nav: React.FC<NavProps> = props => {
...
+  const {
+    state: { themeName },
+  } = useThemeDesigner();
 
-      setName(camelizeName);
-    }
+  // TODO -- fix
+  const handleNameChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    // const newName = e.target.value;
+    //
```
Comment (TristanWatanabe) [inline packages/react-components/theme-designer/src/components/Nav/Nav.tsx:47]: TODO item that still needs to be fixed?

File: packages/react-components/theme-designer/src/components/ColorTokens/AccessiblePairs.ts (hunk 7 above)
Comment (timbellay) [reply packages/react-components/theme-designer/src/components/ColorTokens/AccessiblePairs.ts:13]: Yes we'll keep these commented out for now. Someone will need go through them later and add back in if necessary. Will make a note for that.

File: packages/react-components/theme-designer/src/components/Nav/Nav.tsx (hunk 8 above)
Comment (timbellay) [reply packages/react-components/theme-designer/src/components/Nav/Nav.tsx]: removed

File: packages/react-components/theme-designer/src/colors/palettes.ts (hunk 6 above)
Comment (timbellay) [reply packages/react-components/theme-designer/src/colors/palettes.ts]: rem

File: packages/react-components/theme-designer/src/colors/palettes.ts (hunk 5 above)
Comment (timbellay) [reply packages/react-components/theme-designer/src/colors/palettes.ts]: rem

File: packages/react-components/theme-designer/src/colors/palettes.ts (hunk 4 above)
Comment (timbellay) [reply packages/react-components/theme-designer/src/colors/palettes.ts]: rem

File: packages/react-components/theme-designer/src/colors/palettes.ts (hunk 3 above)
Comment (timbellay) [reply packages/react-components/theme-designer/src/colors/palettes.ts]: rem

File: packages/react-components/theme-designer/src/colors/palettes.ts [hunk 9]
```diff
@@ -14,12 +14,69 @@ import { CurvedHelixPath, Palette, Vec3 } from './types';
...
+
+const snappingPointsForKeyColor = (keyColor: string): number[] => {
+  const hue = hexToHue(keyColor);
+  const range = [
+    hueToSnappingPointsMap[hue][0] * 100,
+    hueToSnappingPointsMap[hue][1] * 100,
+    hueToSnappingPointsMap[hue][2] * 100,
+  ];
+  //   console.log(`hue ${hue} range, ${range}`);
```
[the remark this answers is not in the document]
Comment (timbellay) [reply packages/react-components/theme-designer/src/colors/palettes.ts]: rem

File: packages/react-components/theme-designer/src/colors/palettes.ts (hunk 2 above)
Comment (timbellay) [reply packages/react-components/theme-designer/src/colors/palettes.ts]: rem

File: packages/react-components/theme-designer/src/Context/ThemeDesignerContext.tsx (hunk 1 above)
Comment (timbellay) [reply packages/react-components/theme-designer/src/Context/ThemeDesignerContext.tsx]: changed this to useThemeDesigner

**Review by TristanWatanabe — APPROVED** [review]

File: packages/react-components/theme-designer/src/components/Demo/Demo.tsx [hunk 10]
```diff
@@ -190,8 +184,8 @@ export const Demo: React.FC<ContentProps> = props => {
   const styles = useStyles();
   return (
     <div>
-      <Caption1>Examples</Caption1>
       <div className={mergeClasses(styles.root, props.className)}>
+        {/* <Title3 block>Component Examples</Title3> */}
```
Comment (sopranopillow) [inline packages/react-components/theme-designer/src/components/Demo/Demo.tsx:188]: Is this left by mistake? seems like shouldn't be there.

File: packages/react-components/theme-designer/src/components/Export/ExportPanel.tsx [hunk 11]
```diff
@@ -0,0 +1,185 @@
...
+                  Export Theme
+                </Text>
+                <Button
+                  size="small"
+                  appearance="subtle"
+                  icon={<DismissSquare24Regular />}
+                  // eslint-disable-next-line react/jsx-no-bind
+                  onClick={onCloseExportPanel}
+                />
+              </div>
```
Comment (sopranopillow) [inline packages/react-components/theme-designer/src/components/Export/ExportPanel.tsx:144]: Looks like this button is getting a little messed up:
image

Adding `align-items: center` to the `exportHeader` classname and `height: max-content` to the button's classname should fix it.

File: packages/react-components/theme-designer/src/components/Header/Header.stories.tsx [hunk 12]
```diff
@@ -0,0 +1,3 @@
+import { Header } from './Header';
+export default { component: Header };
+export const Default = {};
```
Comment (sopranopillow) [inline packages/react-components/theme-designer/src/components/Header/Header.stories.tsx:3]: I see you deleted the other stories for the components, is this one needed?

File: packages/react-components/theme-designer/src/components/Sidebar/Form.tsx [hunk 13]
```diff
@@ -43,111 +60,215 @@ const useStyles = makeStyles({
...
+          <AccordionPanel className={styles.accordionContainer}>
+            <div className={styles.labelName}>
+              <Label htmlFor={themeNameInputId}>Theme name</Label>
+              <Input
+                appearance="outline"
+                id={themeNameInputId}
+                onChange={handleThemeNameChange}
+                placeholder={defaultThemePlaceholderName}
+                value={themeName === defaultThemePlaceholderName ? '' : themeName}
+              />
```
Comment (sopranopillow) [inline packages/react-components/theme-designer/src/components/Sidebar/Form.tsx:264]: Not as necessary as the stickers, but would be nice to use field as well.

File: packages/react-components/theme-designer/src/components/Sidebar/Form.tsx [hunk 14]
```diff
@@ -43,111 +60,215 @@ const useStyles = makeStyles({
...
+                <Input
+                  size="small"
+                  type="number"
+                  min={-50}
+                  max={50}
+                  appearance="outline"
+                  id={sidebarId + 'vibrancy input'}
+                  value={vibrancy.toString()}
+                  onChange={handleVibrancyChange}
+                />
```
Comment (sopranopillow) [inline packages/react-components/theme-designer/src/components/Sidebar/Form.tsx:233]: ditto Field usage, for the rest of the file

**Review by sopranopillow — APPROVED** [review]
Mainly minor comments, I would say the sticker ones are needed since those are exposed to the user. The Designer looks great!!!

File: packages/react-components/theme-designer/src/components/Demo/Demo.tsx [hunk 15]
```diff
@@ -144,6 +120,35 @@ export const Column2 = () => {
...
+          <Checkbox label="Option 2" />
+        </div>
+        <div className={styles.controlColumn}>
+          <RadioGroup>
+            <Radio defaultChecked={true} label="Option 1" />
+            <Radio label="Option 2" />
+          </RadioGroup>
+        </div>
+      </div>
+    </div>
```
[the remark this answers is not in the document]
Comment (timbellay) [reply packages/react-components/theme-designer/src/components/Demo/Demo.tsx:147]: I'll add that suggestion as a todo/feedback in the comments. We don't have consensus on the items to show in the sticker sheet.

File: packages/react-components/theme-designer/src/components/Demo/Demo.tsx (hunk 10 above)
Comment (timbellay) [reply packages/react-components/theme-designer/src/components/Demo/Demo.tsx:188]: rem

File: packages/react-components/theme-designer/src/components/Export/ExportPanel.tsx (hunk 11 above)
Comment (timbellay) [reply packages/react-components/theme-designer/src/components/Export/ExportPanel.tsx:144]: Thanks for that suggestion! Changed to "center". Height suggestion to button didn't add anything so didn't add.

File: packages/react-components/theme-designer/src/components/Header/Header.stories.tsx (hunk 12 above)
Comment (timbellay) [reply packages/react-components/theme-designer/src/components/Header/Header.stories.tsx:3]: rem. thanks!

File: packages/react-components/theme-designer/src/components/Sidebar/Form.tsx (hunk 14 above)
Comment (timbellay) [reply packages/react-components/theme-designer/src/components/Sidebar/Form.tsx:233]: Added Field, rem id prop

File: packages/react-components/theme-designer/src/components/Demo/Demo.tsx [hunk 16]
```diff
@@ -96,13 +85,9 @@ const useStyles = makeStyles({
...
   const styles = useStyles();
+  const dropdownId = useId('dropdown-default');
   return (
-    <div className={styles.col1}>
-      <Title3 block>Make an impression</Title3>
-      <Body1 block>
-        Make a big impression with this clean, modern, and mobile-friendly site. Use it to communicate information to
-        people inside or outside your team. Share your ideas, results, and more in this visually compelling format.
-      </Body1>
+    <div className={styles.column}>
```
[the remark this answers is not in the document]
Comment (timbellay) [reply packages/react-components/theme-designer/src/components/Demo/Demo.tsx:90]: changed to suggestion. thanks!

File: packages/react-components/theme-designer/src/components/Sidebar/Form.tsx (hunk 13 above)
Comment (timbellay) [reply packages/react-components/theme-designer/src/components/Sidebar/Form.tsx]: changed

File: packages/react-components/theme-designer/src/components/Header/Header.stories.tsx (hunk 12 above)
Comment (timbellay) [reply packages/react-components/theme-designer/src/components/Header/Header.stories.tsx]: rem. thanks !

**Pull request merged**